### PR TITLE
[RFR] Product and Order admin

### PR DESCRIPTION
**Pull request by Kmaschta**
## TODO
- [x] Move app name into config
- [x] Product admin
- [x] Order admin
- [x] ngAdmin send user token in all requests

## Previews

preview1

preview2

preview3

### CODE REVIEW

File: app/admin/js/main.js [hunk 1]
```diff
@@ -15,21 +15,21 @@ function logout() {
 
 window.logout = logout;
 
-if (!window.sessionStorage.getItem('token')) redirectToLogin();
+// if (!window.sessionStorage.getItem('token')) redirectToLogin();
```
Comment (Kmaschta) [inline app/admin/js/main.js]: I'll remove this in next commit

File: app/admin/js/orders/config.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                .pinned(true)
+                .template('<div class="input-group"><input type="text" ng-model="value" placeholder="Search" class="form-control"></input><span class="input-group-addon"><i class="glyphicon glyphicon-search"></i></span></div>'),
+            // nga.field('customer_id', 'reference')
+            //     .label('Customer')
+            //     .targetEntity(admin.getEntity('customers'))
+            //     .targetField(nga.field('last_name').map((v, e) => e.first_name + ' ' + e.last_name))
+            //     .remoteComplete(true, {
+            //         searchQuery: function(search) { return { q: search }; }
+            //     }),
+            nga.field('status', 'choice')
```
Comment (fzaninotto) [inline app/admin/js/orders/config.js]: I don't believe that this filter works, seeing the related CRUD. Please test.

File: app/admin/js/orders/config.js [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+            nga.field('date', 'datetime'),
+            nga.field('reference', 'reference')
+                .label('Product')
+                .isDetailLink(true)
+                .targetEntity(admin.getEntity('products'))
+                .targetField(nga.field('reference'))
+                .remoteComplete(true, {
+                    searchQuery: (search) => { return {q: search}; },
+                }),
+            // nga.field('customer_id', 'reference')
```
Comment (fzaninotto) [inline app/admin/js/orders/config.js]: commented out code

File: app/admin/js/orders/config.js [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+                .label('Price')
+                .cssClasses('hidden-xs'),
+            nga.field('status'),
+        ])
+        .filters([
+            nga.field('q', 'template')
+                .label('')
+                .pinned(true)
+                .template('<div class="input-group"><input type="text" ng-model="value" placeholder="Search" class="form-control"></input><span class="input-group-addon"><i class="glyphicon glyphicon-search"></i></span></div>'),
+            // nga.field('customer_id', 'reference')
```
Comment (fzaninotto) [inline app/admin/js/orders/config.js]: commented out code

File: app/admin/js/orders/config.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            nga.field('total_gte')
+                .label('Min price'),
+        ])
+        .listActions(['edit', 'delete']);
+
+    orders.creationView()
+        .title('New Order')
+        .fields([
+            nga.field('date', 'datetime'),
+            // nga.field('customer_id', 'reference')
```
Comment (fzaninotto) [inline app/admin/js/orders/config.js]: commented out code

File: app/admin/js/orders/config.js [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+                    { label: 'cancelled', value: 'cancelled' },
+                ])
+                .cssClasses('col-sm-4 col-lg-2'),
+        ]);
+
+    orders.editionView()
+        .title('Order #{{ entry.values.reference }}')
+        .fields([
+            nga.field('date', 'datetime'),
+            // nga.field('customer_id', 'reference')
```
Comment (fzaninotto) [inline app/admin/js/orders/config.js]: commented out code

File: app/admin/js/products/config.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+                .cssClasses('hidden-xs'),
+            nga.field('stock', 'number')
+                .cssClasses('hidden-xs'),
+        ])
+        .filters([
+            nga.field('q', 'template')
+                .label('')
+                .pinned(true)
+                .template('<div class="input-group"><input type="text" ng-model="value" placeholder="Search" class="form-control"></input><span class="input-group-addon"><i class="glyphicon glyphicon-search"></i></span></div>'),
+            nga.field('width_gte', 'number')
```
Comment (fzaninotto) [inline app/admin/js/products/config.js]: Does probably not work

File: config/default.js [hunk 8]
```diff
@@ -3,6 +3,7 @@ var apiUrl = 'http://localhost:' + apiPort; // eslint-disable-line no-var
 var frontendUrl = 'http://localhost:8080'; // eslint-disable-line no-var
 
 module.exports = {
+    appName: 'New App',
```
Comment (alexisjanvier) [inline config/default.js]: Isn't it specific for the admin ? If it is, shouldn't we specify it in a specific admin entry ?
```
module.exports = {
    admin: {
        appName: 'New App',
```

File: config/default.js [hunk 9]
```diff
@@ -3,6 +3,7 @@ var apiUrl = 'http://localhost:' + apiPort; // eslint-disable-line no-var
...
     admin: {
```
Comment (alexisjanvier) [inline config/default.js]: this one ...

File: webpack/plugins.js [hunk 10]
```diff
@@ -1,12 +1,12 @@
-var config = require('config');
-var fs = require('fs');
-var ExtractTextPlugin = require('extract-text-webpack-plugin');
-var HtmlWebpackPlugin = require('html-webpack-plugin');
-var webpack = require('webpack');
+var config = require('config'); // eslint-disable-line no-var
```
Comment (alexisjanvier) [inline webpack/plugins.js]: Why didn't use ES6 here ?

File: config/default.js (hunk 8 above)
Comment (Kmaschta) [inline config/default.js]: For now it is, but the app name can be used everywhere (If it can avoid the use of yeoman or something like that).
But I can change if you want

File: webpack/plugins.js (hunk 10 above)
Comment (djhi) [inline webpack/plugins.js]: I tried once to use es6 in webpack config but it didn't work well at the time. If you know how to make it work, please do so

Comment (Kmaschta) [inline webpack/plugins.js]: No idea, I seen old JS here, I thought that webpack needed it.
But if it works with ES6, yep, let's use it !

**Comment by Kmaschta** [comment]
Waiting for #16

File: webpack/plugins.js (hunk 10 above)
Comment (djhi) [inline webpack/plugins.js]: Will be addressed in another pr ?

Comment (Kmaschta) [inline webpack/plugins.js]: Not in this one, in all cases.

**Comment by alexisjanvier** [comment]
@Kmaschta RFR ?

**Comment by alexisjanvier** [comment]
Nope

**Comment by alexisjanvier** [comment]
Yes :)
